tesseract_time_parameterization/src/time_optimal_trajectory_generation.cpp

```diff
@@ -215,6 +215,122 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(CompositeInstruction& pr
+{
+  if (trajectory.empty())
+    return true;
+
+  // Validate velocity scaling
+  double velocity_scaling_factor = 1.0;
+  if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+  {
+    velocity_scaling_factor = max_velocity_scaling_factor;
+  }
+  else if (max_velocity_scaling_factor == 0.0)
+  {
+    CONSOLE_BRIDGE_logDebug("A max_velocity_scaling_factor of 0.0 was specified, defaulting to %f instead.",
+                            velocity_scaling_factor);
+  }
+  else
+  {
+    CONSOLE_BRIDGE_logWarn("Invalid max_velocity_scaling_factor %f specified, defaulting to %f instead.",
+                           max_velocity_scaling_factor,
+                           velocity_scaling_factor);
+  }
+
+  // Validate acceleration scaling
+  double acceleration_scaling_factor = 1.0;
+  if (max_acceleration_scaling_factor > 0.0 && max_acceleration_scaling_factor <= 1.0)
+  {
+    acceleration_scaling_factor = max_acceleration_scaling_factor;
+  }
+  else if (max_acceleration_scaling_factor == 0.0)
+  {
+    CONSOLE_BRIDGE_logDebug("A max_acceleration_scaling_factor of 0.0 was specified, defaulting to %f instead.",
+                            acceleration_scaling_factor);
+  }
+  else
+  {
+    CONSOLE_BRIDGE_logWarn("Invalid max_acceleration_scaling_factor %f specified, defaulting to %f instead.",
+                           max_acceleration_scaling_factor,
+                           acceleration_scaling_factor);
+  }
+
+  // Validate limits
+  if (max_velocity.rows() != max_acceleration.rows())
+  {
+    CONSOLE_BRIDGE_logError("Invalid max_velocity or max_acceleration specified. They should be the same length");
+  }
+
+  // Flatten program
+  const Eigen::Index num_joints = trajectory.dof();
+  const std::size_t num_points = static_cast<std::size_t>(trajectory.size());
+
+  // This lib does not actually work properly when angles wrap around, so we need to unwind the path first
+  //  trajectory.unwind(); /// @todo
+
+  // Have to convert into Eigen data structs and remove repeated points
+  //  (https://github.com/tobiaskunz/trajectories/issues/3)
+  std::list<Eigen::VectorXd> points;
+  for (Eigen::Index p = 0; p < num_points; ++p)
+  {
+    const Eigen::VectorXd& position = trajectory.getPosition(p);
+    bool diverse_point = (p == 0);
+
+    for (Eigen::Index j = 0; j < num_joints; j++)
+    {
+      if (p > 0 && std::abs(position[j] - points.back()[j]) > min_angle_change_)
+        diverse_point = true;
+    }
+
+    if (diverse_point)
+      points.push_back(position);
+  }
+
+  // Return trajectory with only the first waypoint if there are not multiple diverse points
+  if (points.size() == 1)
+  {
+    CONSOLE_BRIDGE_logDebug("Trajectory is parameterized with 0.0 dynamics since it only contains a single distinct "
+                            "waypoint.");
+    trajectory.setData(0, Eigen::VectorXd::Zero(num_joints), Eigen::VectorXd::Zero(num_joints), 0);
```

Comment:
So this only adds time param for the first point. The old version I think only returned a single point. Perhaps we should just set the time to all 0?

Reply:
Yea that would be good. I have made the fix.